### PR TITLE
add template parameters to publishers

### PR DESCRIPTION
**Pull request by jacquelinekay**
Connects to ros2/rclcpp#121

### CODE REVIEW

**Comment by esteve** [comment]
+1 (pending CI)

File: intra_process_demo/src/cyclic_pipeline/cyclic_pipeline.cpp [hunk 1]
```diff
@@ -30,19 +30,19 @@ struct IncrementerPipe : public rclcpp::Node
...
-        }
-        printf("  done.\n");
-        msg->data++;  // Increment the message's data.
-        printf("Incrementing and sending with value: %d, and address: %p\n", msg->data, msg.get());
-        this->pub->publish(msg);  // Send the message along to the output topic.
-      });
+      printf("Received message with value:         %d, and address: %p\n", msg->data, msg.get());
+      printf("  sleeping for 1 second...\n");
+      if (!rclcpp::sleep_for(1_s)) {
+        return;    // Return if the sleep failed (e.g. on ctrl-c).
```
Comment (esteve) [inline intra_process_demo/src/cyclic_pipeline/cyclic_pipeline.cpp]: There seems to be tab in here and in lines 39 and 41.

File: intra_process_demo/src/image_pipeline/watermark_node.hpp [hunk 2]
```diff
@@ -35,22 +35,22 @@ class WatermarkNode : public rclcpp::Node
...
-          msg->width, msg->height,
-          encoding2mat_type(msg->encoding),
-          msg->data.data());
-        // Annotate the image with the pid, pointer address, and the watermark text.
-        std::stringstream ss;
-        ss << "pid: " << GETPID() << ", ptr: " << msg.get() << " " << text;
-        draw_on_image(cv_mat, ss.str(), 40);
-        this->pub_->publish(msg);  // Publish it along.
-      });
+        [this, text](sensor_msgs::msg::Image::UniquePtr & msg) {
```
Comment (esteve) [inline intra_process_demo/src/image_pipeline/watermark_node.hpp]: Tab here and in line 48.

File: intra_process_demo/src/cyclic_pipeline/cyclic_pipeline.cpp (hunk 1 above)
Comment (jacquelinekay) [inline intra_process_demo/src/cyclic_pipeline/cyclic_pipeline.cpp]: oh jeez. This was all uncrustify's fault

**Comment by jacquelinekay** [comment]
I reverted the whitespace-only changes.

**Pull request merged**